src/rawposix/src/sys_calls.rs

```diff
@@ -0,0 +1,1161 @@
+/// * `which_arg` / `which_arg_cageid` – Encoded argument specifying which timer to use
+///   (`ITIMER_REAL`, `ITIMER_VIRTUAL`, `ITIMER_PROF`). Only `ITIMER_REAL` is implemented.
+/// * `new_value_arg` / `new_value_arg_cageid` – Pointer to a new `itimerval` struct.
+///   If non-null, this specifies the new timer settings.
+/// * `old_value_arg` / `old_value_arg_cageid` – Pointer to an `itimerval` struct
+///   where the previous timer value should be stored. If non-null, the current
+///   timer is copied here before being changed.
+/// 
+/// ## Returns
+/// * `0` on success.
+/// * Negative errno (`EFAULT`, etc.) on failure.
+pub fn setitimer_syscall(
+    cageid: u64,
+    which_arg: u64, 
+    which_arg_cageid: u64,
+    new_value_arg: u64, 
+    new_value_arg_cageid: u64,
+    old_value_arg: u64, 
+    old_value_arg_cageid: u64,
+    arg4: u64, 
+    arg4_cageid: u64,
+    arg5: u64, 
+    arg5_cageid: u64,
+    arg6: u64, 
+    arg6_cageid: u64,
+) -> i32 {
+    let which = sc_convert_sysarg_to_i32(which_arg, which_arg_cageid, cageid);
+    let new_value = sc_convert_itimerval(new_value_arg, new_value_arg_cageid, cageid);
+    let old_value = sc_convert_itimerval_mut(old_value_arg, old_value_arg_cageid, cageid);
+    // Validate that extra arguments are indeed unused.
+    if !(sc_unusedarg(arg4, arg4_cageid)
+         && sc_unusedarg(arg5, arg5_cageid)
+         && sc_unusedarg(arg6, arg6_cageid)) {
+        return syscall_error(Errno::EFAULT, "setitimer", "Invalid extra arguments");
+    }
+
+    // get the cage instance
+    let cage = get_cage(cageid).unwrap();
+
+    match which {
+        ITIMER_REAL => {
+            if let Some(some_old_value) = old_value {
+                let (curr_duration, next_duration) = cage.interval_timer.get_itimer();
+                some_old_value.it_value.tv_sec = curr_duration.as_secs() as i64;
+                some_old_value.it_value.tv_usec = curr_duration.subsec_millis() as i64;
+                some_old_value.it_interval.tv_sec = next_duration.as_secs() as i64;
+                some_old_value.it_interval.tv_usec = next_duration.subsec_millis() as i64;
+            }
+
+            if let Some(some_new_value) = new_value {
+                let curr_duration = Duration::new(
+                    some_new_value.it_value.tv_sec as u64,
+                    some_new_value.it_value.tv_usec as u32,
+                );
+                let next_duration = Duration::new(
+                    some_new_value.it_interval.tv_sec as u64,
+                    some_new_value.it_interval.tv_usec as u32,
+                );
+
+                cage.interval_timer.set_itimer(curr_duration, next_duration);
+            }
+        }
+
+        _ => { /* ITIMER_VIRTUAL and ITIMER_PROF is not implemented*/ }
+    }
+    0
+}
+
+/// Those functions are required by wasmtime to create the first cage. `verbosity` indicates whether
+/// detailed error messages will be printed if set
+pub fn lindrustinit(verbosity: isize) {
```

Comment:
We really should change these function names (these are originally my fault in rustposix I believe).

I think just rawposix_init/fini is better?